crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -560,12 +575,16 @@ async fn execute_ics20_transfer_bridge_lock<S: StateWriteExt>(
         return Ok(());
     }
 
+    // assert memo is valid
+    let deposit_memo: Ics20TransferDepositMemo =
+        serde_json::from_str(&memo).context("failed to parse memo as Ics20TransferDepositMemo")?;
+
     ensure!(
-        !destination_address.is_empty(),
+        !deposit_memo.rollup_address.is_empty(),
         "packet memo field must be set for bridge account recipient",
     );
```

Comment:
We can probably add additional validation on length? Max 256 length?

Reply:
since we already check the entire packet length, additional validation here just put constraints on the rollup address format. however we can probably assume rollup addresses aren't >256 bytes so will add